docs/sdk/src/reference_docs/chain_spec_runtime/tests/chain_spec_builder_tests.rs

```diff
@@ -8,185 +9,171 @@ fn wasm_file_path() -> &'static str {
 
 const CHAIN_SPEC_BUILDER_PATH: &str = "../../../../../target/release/chain-spec-builder";
 
+macro_rules! bash(
+	( chain-spec-builder $x:tt, $($a:tt)* ) => {{
+		let path = get_chain_spec_builder_path();
+		let x: serde_json::Value = json!($x);
+		let output = spawn_with_output!(
+			$path $($a)*
+		)
+		.expect("a process running. qed")
+		.wait_with_output()
+		.expect("to get output. qed.");
+		let output0: serde_json::Value = serde_json::from_slice(&output.as_bytes()).unwrap();
+		assert_eq!(output0, x, "Output did not match expected");
+	}}
+);
+
+macro_rules! bash2(
+	( chain-spec-builder $x:tt, $($a:tt)* ) => {{
+		let path = get_chain_spec_builder_path();
+		let x: serde_json::Value = json!($x);
+		let output = spawn_with_output!(
+			$path $($a)*
+		)
+		.expect("a process running. qed")
+		.wait_with_output()
+		.expect("to get output. qed.");
+		let mut output0: serde_json::Value = serde_json::from_slice(&output.as_bytes()).unwrap();
+		//remove code field for better readability
+	if let Some(code) = output0["genesis"]["runtimeGenesis"].as_object_mut().unwrap().get_mut("code")
+	{
+		*code = Value::String("0x123".to_string());
+	}
+		assert_eq!(output0, x, "Output did not match expected");
+	}}
+);
+
 fn get_chain_spec_builder_path() -> &'static str {
-	// dev-dependencies do not build binary. So let's do the naive work-around here:
-	let _ = std::process::Command::new("cargo")
-		.arg("build")
-		.arg("--release")
-		.arg("-p")
-		.arg("staging-chain-spec-builder")
-		.arg("--bin")
-		.arg("chain-spec-builder")
-		.status()
-		.expect("Failed to execute command");
+	run_cmd!(
+		cargo build --release -p staging-chain-spec-builder --bin chain-spec-builder
+	)
+	.expect("Failed to execute command");
 	CHAIN_SPEC_BUILDER_PATH
 }
 
-#[test]
 #[docify::export]
-fn list_presets() {
-	let output = Command::new(get_chain_spec_builder_path())
-		.arg("list-presets")
-		.arg("-r")
-		.arg(wasm_file_path())
-		.output()
-		.expect("Failed to execute command");
+fn cmd_list_presets(path: &str) {
+	bash!(
+		chain-spec-builder {
+			"presets":[
+				"preset_1",
+				"preset_2",
+				"preset_3",
+				"preset_4",
+				"preset_invalid"
+			]
+		}, list-presets -r $path
+	);
```

Comment:
I have two remarks that apply to all the tests:

1. Can we not assert inside the `bash!` macro? 
2. Can we use `#[docify::export_content]` instead of `#[docify::export]`? In this way we'll export only the function content, ignoring its signature.

The idea is to assert in the test which calls this function (e.g. `list_presets()`), and the function to return a `String` that can be parsed as JSON accordingly. In the end the command we'll appear in the docs like:
```rust
bash!(
    chain-spec-builder list-presets -r $RUNTIME_PATH
)
```

LMK if this doesn't make sense.

Reply:
@iulianbarbu , I really did the minimum for `your_first_node.rs`. please let me know if it's not satisfying adn I will grind a bit more. For the rest, all corrections have been implemented, and the CI seems to be happy.